packages/mui-material/src/styles/CssVarsProvider.tsx

```diff
@@ -36,11 +36,9 @@ const {
   },
 });
 
-let warnedOnce = false;
-
-// TODO: remove in v7
 // eslint-disable-next-line @typescript-eslint/naming-convention
 function Experimental_CssVarsProvider(props: any) {
+  const [warnedOnce, setWarnedOnce] = React.useState(false);
```

Comment:
Let's use this other approach that we already use in other places https://github.com/mui/material-ui/blob/043bb210c12091ce3373addb84e3dea52216bc33/packages/mui-lab/src/Alert/Alert.js#L5-L26

Reply:
In recent updates from next branch this file seems to be deleted.